src/GatewayAPI/Database/TransactionQuerier.cs

```diff
@@ -332,37 +344,113 @@ private async Task<List<long>> GetAccountTransactionStateVersions(AccountTransac
 
     private async Task<List<Gateway.TransactionInfo>> GetTransactions(List<long> transactionStateVersions)
     {
-        var transactionWithOperationGroups = await _dbContext.LedgerTransactions
-            .Where(t => transactionStateVersions.Contains(t.ResultantStateVersion))
-            .Include(t => t.SubstantiveOperationGroups)
-            .ThenInclude(op => op.InferredAction!.Resource)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.Validator)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.FromAccount)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.ToAccount)
-            .Include(t => t.RawTransaction)
-            .OrderByDescending(lt => lt.ResultantStateVersion)
-            .AsSplitQuery() // See https://docs.microsoft.com/en-us/ef/core/querying/single-split-queries
-            .ToListAsync();
+        async Task<List<LedgerTransaction>> LoadLedgerTransactions(List<long> ids)
+        {
+            await using var localCtx = await _dbContextFactory.CreateDbContextAsync();
+
+            // we're using explicit transaction here to ensure that:
+            // a) SET LOCAL call does not leak outside of this transaction
+            // b) EF will actually reuse same database connection for both queries
+            await using var tx = await localCtx.Database.BeginTransactionAsync();
+
+            // query below tend to operate on quite significant volume of data (over 4 MiB)
+            // so in order to prevent HDD/SSD usage while applying ORDER BY clause
+            await localCtx.Database.ExecuteSqlRawAsync("SET LOCAL work_mem = '32MB';");
+
+            var result = await localCtx.LedgerTransactions
```

Comment:
It will likely be even more performant to separate the LedgerTransactions query from the OperationGroups query, query them sequentially, and also join them together manually.

That way, we get rid of any joins and duplicated data.

I don't know if this is worth it or not? Depends on how much the duplicated transaction data in the join is costing VS whether adding an extra DB round trip makes the response too slow.

Reply:
Split query might also do this properly, but maybe not?

Reply:
I doubt it would make any significant change. There's no “cartesian explosion” risk here and the volume of repeated data coming from `ledger_transactions` isn't probably worth extra network roundtrip. I might evaluate that if you insist though.

Reply:
Happy to go with your gut on this, but also, it'll probably be 5 mins to try it out and see... so up to you :)

Reply:
BTW another thought I had is if removing the order by here will make it more performant? As we can order on the client side.